tailcfg/tailcfg.go

```diff
@@ -3185,23 +3185,23 @@ type PeerChange struct {
 	Endpoints []netip.AddrPort `json:",omitempty"`
 
 	// Key, if non-nil, means that the NodeID's wireguard public key changed.
-	Key *key.NodePublic `json:",omitempty"`
+	Key *key.NodePublic `json:",omitzero"`
```

Comment:
I just want to call out a subtle semantic change switching `omitempty` to `omitzero` on `*T` where `T.IsZero` exists. Under the semantics of the `omitzero`, the `IsZero` method will be consulted and omit the field. Thus, `&key.NodePublic{}` will be omitted in addition to nil. This subtle difference is best [demonstrated through a an example](https://go.dev/play/p/zjcuQx5eYWM).

In many cases, this may be safe since we don't care about distinguishing between a nil `*T` versus a zero `T`. In fact, most cases where `omitempty` on `*T` existed because `omitempty` didn't work for struct types, so a pointer was the hacky workaround.